File: sdk-server/modules/sys/models/admin_role.go
package models

import (
	"dilu/common/model"
	"time"

	"gorm.io/gorm"
)

// 角色
type AdminRole struct {
	Id        int            `json:"id" gorm:"type:int unsigned;primaryKey;autoIncrement;comment:主键"` //主键
	Name      string         `json:"name" gorm:"type:varchar(128);comment:角色名称"`                      //角色名称
	RoleKey   string         `json:"roleKey" gorm:"type:varchar(128);comment:角色代码"`                   //角色代码
	RoleSort  uint           `json:"roleSort" gorm:"type:int unsigned;comment:排序"`                    //排序
	Status    int8           `json:"status" gorm:"type:tinyint;comment:状态"`                           //状态
	Remark    string         `json:"remark" gorm:"type:varchar(255);comment:备注"`                      //备注
	MenuIds   model.Array    `json:"menuIds" gorm:"type:json;comment:菜单ID"`                           //菜单ID
	CreateBy  int            `json:"createBy" gorm:"type:int unsigned;comment:创建者"`                   //创建者
	UpdateBy  int            `json:"updateBy" gorm:"type:int unsigned;comment:更新者"`                   //更新者
	CreatedAt time.Time      `json:"createdAt" gorm:"type:datetime(3);comment:创建时间"`                  //创建时间
	UpdatedAt time.Time      `json:"updatedAt" gorm:"type:datetime(3);comment:最后更新时间"`                //最后更新时间
	DeletedAt gorm.DeletedAt `json:"-" gorm:"index;comment:删除时间"`                                     //删除时间
}

const TBAdminRole = "admin_role"

func (AdminRole) TableName() string {
	return TBAdminRole
}

File: sdk-admin/src/api/sys/admin-role.ts
import { http } from "@/utils/http";
import { Result, PageResult } from "@/api/common";
//import { AdminRole } from "@/api/sys/admin-role.d";

/** 获取角色管理列表 */
export const getAdminRolePage = (data?: object) => {
  return http.request<Result<PageResult<AdminRole>>>(
    "post",
    "/api/sys/admin-role/page",
    {
      data
    }
  );
};

export const getAdminRoleList = (data?: object) => {
  return http.request<Result<Array<AdminRole>>>(
    "post",
    "/api/sys/admin-role/list",
    {
      data
    }
  );
};

/** 获取角色 */
export const getAdminRole = (data?: object) => {
  return http.request<Result<AdminRole>>("post", "/api/sys/admin-role/get", {
    data
  });
};

/** 创建角色 */
export const createAdminRole = (data?: object) => {
  return http.request<Result<AdminRole>>("post", "/api/sys/admin-role/create", {
    data
  });
};

/** 更新角色 */
export const updateAdminRole = (data?: object) => {
  return http.request<Result<AdminRole>>("post", "/api/sys/admin-role/update",{
    data
  });
};

/** 删除角色 */
export const delAdminRole = (data?: object) => {
  return http.request<Result<AdminRole>>("post", "/api/sys/admin-role/del",{
    data
  });
};

//Model

/** 角色 */
interface AdminRole {
  /** 主键 */
  id: number;
  /** 角色名称 */
  name: string;
  /** 角色代码 */
  roleKey: string;
  /** 排序 */
  roleSort: number;
  /** 状态 */
  status: number;
  /** 备注 */
  remark: string;
  /** 创建者 */
  createBy: number;
  /** 更新者 */
  updateBy: number;
  /** 创建时间 */
  createdAt: Date;
  /** 最后更新时间 */
  updatedAt: Date;
  /** 删除时间 */
  deletedAt: Date;
}

interface AdminRoleFormItemProps {
  /** 主键 */
  id: number;
  /** 角色名称 */
  name: string;
  /** 角色代码 */
  roleKey: string;
  /** 排序 */
  roleSort: number;
  /** 状态 */
  status: number;
  /** 备注 */
  remark: string;
}
interface AdminRoleFormProps {
  formInline: AdminRoleFormItemProps;
}

export type { AdminRoleFormItemProps, AdminRoleFormProps, AdminRole };
